packages/discovery-provider/src/tasks/entity_manager/entities/dashboard_wallet_user.py

```diff
@@ -0,0 +1,224 @@
+            raise IndexingValidationError(
+                f"Invalid Delete Dashboard Wallet User Transaction, dashboard wallet user with wallet {dashboard_wallet} does not exist"
+            )
+        # Either the user or the dashboard wallet can sign the Delete tx
+        if not user_matches_signer and not params.signer.lower() == dashboard_wallet:
+            raise IndexingValidationError(
+                f"Invalid Delete Dashboard Wallet User Transaction, signature does not match user or dashboard wallet"
+            )
+    elif params.action == Action.CREATE:
+        if not user_matches_signer:
+            raise IndexingValidationError(
+                f"Invalid Create Dashboard Wallet User Transaction, signature does not match user"
+            )            
+        if not metadata["wallet_signature"]:
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet User Transaction, wallet signature is required and was not provided"
+            )
+        if dashboard_wallet in params.existing_records["DashboardWalletUser"] and params.existing_records["DashboardWalletUser"][dashboard_wallet].is_delete == False:
+            raise IndexingValidationError(
+                f"Invalid Create Dashboard Wallet User Transaction, dashboard wallet {dashboard_wallet} already has an assigned user"
+            )
+        # Expect wallet_signature message to be "Connecting Audius user id {user hash id} at {timestamp}"
+        if (
+            not isinstance(metadata["wallet_signature"], dict)
+            or not metadata["wallet_signature"]
+            .get("message", "")
+            .startswith("Connecting Audius user id")
+            or not matches_user_id((metadata["wallet_signature"].get("message", "").split()[-3]), user_id)
+            or not is_within_5_minutes(
+                (metadata["wallet_signature"].get("message", "").split())[-1]
+            )
+        ):
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet Transaction, wallet signature provided does not have correct message"
+            )
+        try:
+            signature_address = get_address_from_signature(metadata["wallet_signature"])
+        except:
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet User Transaction, signature provided is invalid"
+            )
+        if not signature_address or not signature_address.lower() == dashboard_wallet:
+            raise IndexingValidationError(
+                "Invalid Create Dashboard Wallet User Transaction, signature provided is invalid"
+            )
+    else:
+        raise IndexingValidationError(
+            f"Invalid Dashboard Wallet User Transaction, action {params.action} is not valid"
+        )
+
+
+def create_dashboard_wallet_user(params: ManageEntityParameters):
+    metadata = get_create_dashboard_wallet_user_metadata_from_raw(params.metadata)
```

Comment:
does it make sense for the "validation" to be part of the creation? if the pattern is already established with other entities to do this way, all good just my thoughts. the idea is that at least all Indexing Errors are addressed in the helper functions, so the create step can just be extra simple